common/types_spec.go

```diff
@@ -40,211 +47,393 @@ func BuildGetHeaderResponse(payload *BuilderSubmitBlockRequest, sk *bls.SecretKe
+			StateRoot:     blindedBlock.Message.StateRoot,
+			Body: &capella.BeaconBlockBody{
+				BLSToExecutionChanges: blindedBlock.Message.Body.BLSToExecutionChanges,
+				RANDAOReveal:          blindedBlock.Message.Body.RANDAOReveal,
+				ETH1Data:              blindedBlock.Message.Body.ETH1Data,
+				Graffiti:              blindedBlock.Message.Body.Graffiti,
+				ProposerSlashings:     blindedBlock.Message.Body.ProposerSlashings,
+				AttesterSlashings:     blindedBlock.Message.Body.AttesterSlashings,
+				Attestations:          blindedBlock.Message.Body.Attestations,
+				Deposits:              blindedBlock.Message.Body.Deposits,
+				VoluntaryExits:        blindedBlock.Message.Body.VoluntaryExits,
+				SyncAggregate:         blindedBlock.Message.Body.SyncAggregate,
+				ExecutionPayload:      executionPayload,
 			},
-			Capella: nil,
-		}, nil
+		},
 	}
+}
 
-	if payload.Capella != nil {
-		return &GetPayloadResponse{ //nolint:exhaustruct
-			Capella: &api.VersionedExecutionPayload{ //nolint:exhaustruct
-				Version: consensusspec.DataVersionCapella,
-				Capella: payload.Capella.ExecutionPayload,
+func DenebUnblindSignedBlock(blindedBlock *eth2ApiV1Deneb.SignedBlindedBeaconBlock, blockPayload *builderApiDeneb.ExecutionPayloadAndBlobsBundle, blockRoot phase0.Root) *eth2ApiV1Deneb.SignedBlockContents {
+	return &eth2ApiV1Deneb.SignedBlockContents{
+		SignedBlock: &deneb.SignedBeaconBlock{
+			Message: &deneb.BeaconBlock{
+				Slot:          blindedBlock.Message.Slot,
+				ProposerIndex: blindedBlock.Message.ProposerIndex,
+				ParentRoot:    blindedBlock.Message.ParentRoot,
+				StateRoot:     blindedBlock.Message.StateRoot,
+				Body: &deneb.BeaconBlockBody{
+					RANDAOReveal:          blindedBlock.Message.Body.RANDAOReveal,
+					ETH1Data:              blindedBlock.Message.Body.ETH1Data,
+					Graffiti:              blindedBlock.Message.Body.Graffiti,
+					ProposerSlashings:     blindedBlock.Message.Body.ProposerSlashings,
+					AttesterSlashings:     blindedBlock.Message.Body.AttesterSlashings,
+					Attestations:          blindedBlock.Message.Body.Attestations,
+					Deposits:              blindedBlock.Message.Body.Deposits,
+					VoluntaryExits:        blindedBlock.Message.Body.VoluntaryExits,
+					SyncAggregate:         blindedBlock.Message.Body.SyncAggregate,
+					ExecutionPayload:      blockPayload.ExecutionPayload,
+					BLSToExecutionChanges: blindedBlock.Message.Body.BLSToExecutionChanges,
+					BlobKZGCommitments:    blindedBlock.Message.Body.BlobKZGCommitments,
+				},
 			},
-		}, nil
+			Signature: blindedBlock.Signature,
+		},
+		KZGProofs: blockPayload.BlobsBundle.Proofs,
+		Blobs:     blockPayload.BlobsBundle.Blobs,
 	}
+}
 
-	return nil, ErrEmptyPayload
+type BuilderBlockValidationRequest struct {
+	*VersionedSubmitBlockRequest
+	RegisteredGasLimit    uint64       `json:"registered_gas_limit,string"`
+	ParentBeaconBlockRoot *phase0.Root `json:"parent_beacon_block_root,omitempty"`
 }
```

Comment:
Will the validator function allow `ParentBeaconBlockRoot` to be missing? It doesn't appear to be an optional field in the builder's [`BuilderBlockValidationRequestV3`](https://github.com/flashbots/builder/blob/ae7357b26b3c6bf6e0bb97bf6669e62359854339/eth/block-validation/api.go#L193-L197) struct. Should `BuilderBlockValidationRequest` have the `omitempty` qualifier?

<img width="598" alt="image" src="https://github.com/flashbots/mev-boost-relay/assets/95511699/6bf0c03a-2cb3-45ee-b686-596811bc6baa">

Reply:
I think the builder's BuilderBlockValidationRequestV3 struct will unmashal to a zero value (0 hash) as it does not validate that it is a required field.